rs/bitcoin/ckbtc/minter/src/state.rs

```diff
@@ -997,67 +998,71 @@ impl CkBtcMinterState {
         }
     }
 
-    pub fn has_ignored_utxo(&self, utxo: &Utxo) -> bool {
-        self.ignored_utxos.contains(utxo)
+    /// Adds given UTXO to the set of discarded UTXOs.
+    pub fn discard_utxo(&mut self, utxo: Utxo, account: Account, reason: DiscardedReason) -> bool {
+        self.ensure_reason_consistent_with_state(&utxo, reason);
+        self.discarded_utxos.insert(account, utxo, reason)
     }
 
-    /// Adds given UTXO to the set of ignored UTXOs.
-    fn ignore_utxo(&mut self, utxo: Utxo) {
-        assert!(utxo.value <= self.kyt_fee);
-        self.ignored_utxos.insert(utxo);
+    #[deprecated(note = "Use discard_utxo() instead")]
+    pub fn discard_utxo_without_account(&mut self, utxo: Utxo, reason: DiscardedReason) {
+        self.ensure_reason_consistent_with_state(&utxo, reason);
+        #[allow(deprecated)]
+        self.discarded_utxos.insert_without_account(utxo, reason)
     }
 
-    /// Marks the given UTXO as checked.
-    /// If the UTXO is clean, we increase the owed KYT amount if there is a KYT provider, and
+    fn ensure_reason_consistent_with_state(&self, utxo: &Utxo, reason: DiscardedReason) {
+        match reason {
+            DiscardedReason::ValueTooSmall => {
+                assert!(utxo.value <= self.kyt_fee);
+            }
+            DiscardedReason::Quarantined => {}
+        }
+    }
+
+    /// Marks the given UTXO as successfully checked.
+    /// We increase the owed KYT amount if there is a KYT provider, and
```

Comment:
Just to be clear, is the "KYT provider" the KYT canister or the API key provider?
The latter doesn't exist anymore, so there is no need to keep track of any owed amount, right?

Reply:
I think we need to keep these code because the historical events require them, and the owed amounts are also used in consistency check.